daprdocs/content/en/js-sdk-docs/js-client/_index.md

```diff
@@ -294,14 +304,114 @@ async function start() {
+  await server.pubsub.subscribe(pubSubName, topic, async (data: any) => console.log(`Received Data: ${JSON.stringify(data)}`));
 
+  // Method 2: Direct susbcription with options through the `subscribeWithOptions` method
+  await server.pubsub.subscribeWithOptions(pubSubName, topic, { 
+    callback: async (data: any) => console.log(`Received Data: ${JSON.stringify(data)}`) 
+  });
+
+  // Method 3: Subscription afterwards through the `susbcribeOnEvent` method
+  await server.pubsub.subscribeWithOptions('pubsub-redis', 'topic-options-1', {});
+  server.pubsub.subscribeOnEvent("pubsub-redis", "topic-options-1", "default", async (data) => { console.log(`Received Data: ${JSON.stringify(data)}`) });
+
+  // Start the server
   await server.start();
 }
 ```
 
 > For a full list of state operations visit [How-To: Publish & subscribe]({{< ref howto-publish-subscribe.md >}}).
 
+#### Subscribe to messages rule based
+
+Dapr [supports routing messages](https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-route-messages/
+) to different handlers (routes) based on rules. 
+
+> E.g., think of that you are writing an application that needs to handle messages depending on their "type" with Dapr you can then route it to `handlerType1` and `handlerType2` with a default being `handlerDefault`
+
+```javascript
+import { DaprServer } from "@dapr/dapr";
+
+const daprHost = "127.0.0.1"; // Dapr Sidecar Host
+const daprPort = "3500"; // Dapr Sidecar Port of this Example Server
+const serverHost = "127.0.0.1"; // App Host of this Example Server
+const serverPort = "50051"; // App Port of this Example Server "
+
+async function start() {
+  const server = new DaprServer(serverHost, serverPort, daprHost, daprPort);
+
+  const pubSubName = "my-pubsub-name";
+  const topic = "topic-a";
+
+  // Configure Subscriber for a Topic with rule set
+  await server.pubsub.subscribe('pubsub-redis', 'topic-1', {
+    default: "/default",
+    rules: [
+      {
+        match: `event.type == "my-type-1"`,
+        path: "/type-1"
+      },
+      {
+        match: `event.type == "my-type-2"`,
+        path: "/type-2"
+      }
+    ]
+  });
+
+  // Add handlers for each route
+  server.pubsub.subscribeOnEvent("pubsub-redis", "topic-1", "default", async (data) => { console.log(`Handling Default`) });
+  server.pubsub.subscribeOnEvent("pubsub-redis", "topic-1", "type-1", async (data) => { console.log(`Handling Type 1`) });
+  server.pubsub.subscribeOnEvent("pubsub-redis", "topic-1", "type-2", async (data) => { console.log(`Handling Type 2`) });
+
+  // Start the server
+  await server.start();
+}
+```
+
+#### DeadLetter support
+
+As per https://docs.dapr.io/developing-applications/building-blocks/pubsub/pubsub-deadletter/ Dapr supports deadletter messages. This means that when a message fails to be processed, it gets sent to a deadletter portion of the queue. E.g., when a message fails to be handled on `/my-queue` it will be sent to `/my-queue-failed`.
```

Comment:
```suggestion
As per https://docs.dapr.io/developing-applications/building-blocks/pubsub/pubsub-deadletter/ Dapr supports deadletter messages. This means that when a message fails to be processed, it gets sent to a deadletter portion of the queue. E.g., when a message fails to be handled on `/my-queue` it will be sent to `/my-queue-failed`.
```
```suggestion
Dapr supports [dead letter topic](https://docs.dapr.io/developing-applications/building-blocks/pubsub/pubsub-deadletter/). This means that when a message fails to be processed, it gets sent to a deadletter portion of the queue. E.g., when a message fails to be handled on `/my-queue` it will be sent to `/my-queue-failed`.
```